draft-bbs-signatures.md

```diff
@@ -388,80 +390,77 @@ Procedure:
 
 ### Sign
 
-Sign computes a signature from SK, PK, over a vector of messages. This method
-describes deterministic signing. For threshold signing, XOF can be replaced
-with a PRF due to the insecurity of deterministic threshold signing.
+This operation computes a deterministic signature from a secret key (SK), over a header and a vector of messages.
 
 ```
-signature = Sign(SK, PK, (msg_1,..., msg_L), (H_1,..., H_L), header)
+signature = Sign(SK, header, (msg_1,..., msg_L), (H_1,..., H_L))
 
 Inputs:
 
-- msg_1,..., msg_L, octet strings. Messages to be signed.
-- H_1,..., H_L, points of G1. Generators used to sign the messages.
-- SK, a secret key output from KeyGen.
-- PK, a public key output from SkToPk.
-- header, an optional octet string containing context and application specific
-          information. If not supplied, it defaults to an empty string.
+- SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
+- header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
+- msg_1,..., msg_L (OPTIONAL), a vector of octet strings. Application specific messages.
+- H_1,..., H_L (OPTIONAL), points of G1. Generators used to commit each message.
 
 Parameters:
 
-- Ciphersuite_ID, octet string. The unique ID of the ciphersuite.
-- H_s, point of G1. The generator for the blinding value of the signature.
-- H_d, point of G1. The generator used to sign the signature domain.
+- Ciphersuite_ID (REQUIRED), ASCII string. The unique ID of the ciphersuite.
+- H_s (REQUIRED), point of G1. The generator for the blinding value of the signature.
+- H_d (REQUIRED), point of G1. The generator used to sign the signature domain.
 
 Outputs:
 
 - signature, an octet string.
 
 Procedure:
 
-1. (W, H0, H) = octets_to_point(PK)
+1. PK = SkToPk(SK)
```

Comment:
Although i like the simpler API with not having to supply the PK this may add unnecessary computations?? The PK will most likely be cached. It doesn't really guard against un trusted issuers either (they can just skip this step).

Reply:
Im happy to keep supplying the public key to the operation API, provided we add a consideration that implementations SHOULD validate the relationship between SK and PK, does that work as a compromise?

Reply:
If we are worried about issuers making mistakes then yes a consideration sounds as a better option. Maybe another option could also be to make the relationship clear in the definition of the inputs in Sign? Something like;

```
- SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
- PK (REQUIRED), an octet string encoded public key output from SkToPk on input the above SK.
```